lib/ui/shop/shop_product_range_page.dart

```diff
@@ -237,6 +234,48 @@ class _ShopProductRangePageState extends PageStatePlante<ShopProductRangePage> {
     );
   }
 
+  FutureBuilder<String> _showSuggestedProductTitle() {
+    return FutureBuilder(
```

Comment:
ok let me have a look. although i am confused now how to use futurebuilder... because the future is outside the build?

Reply:
In our case each time `buildPage` is called `_getSuggestProductsStoreLabel` also gets called, which means a new future is created for each of the page's rebuilds.
I'm not sure if we need futurebuilder at all - the app worked fine without it in the past, and its first usage seems like futurebuilder overcomplicates things a little.

Reply:
i don't think futurebuilder is complicated :-). Personally i think having a statefull widget is more complicated just to handle a future :-). But you are more experienced and I agree that it would be better to load the title only once. I'll try a refactor with Riverpod.